src/js/network-vis.js

```diff
@@ -232,28 +232,38 @@ const renderNetwork = (dataNodes, dataEdges) => {
       });
       edges.update(oldClickedEdgeIds);
     }
+    oldClickedNodeIds = [];
+    oldClickedEdgeIds = [];
   };
 
-  const highlightNodes = (nodes) => {
-    clearColorSelection();
+  const highlightNodes = (nodes, clearSelection = true) => {
+    if (clearSelection) clearColorSelection();
     let nodIds = [];
 
     nodes.map((node) => {
-      nodIds.push(node.name);
+      nodIds.push(node.id);
     });
     network.selectNodes(nodIds);
   };
+  //TODO: consolidate with network.on click handler
+  const nodeSelectionEventHandler = (nodeId, connectedNodes) => {
```

Comment:
Please comment your function definitions

Also, please leave an empty space above your function definitions. This bunched up formatting is terrible to read

Reply:
Done